decimalSuite_test.go

```diff
@@ -93,32 +129,45 @@ func isRoundingErr(res, expected Decimal64) bool {
-func getInput(file string) (data []testCaseStrings) {
-	r := regexp.MustCompile(`(?:\n)` + // start with newline (?: non capturing group)
+func getInput(line string) (testCaseStrings, error) {
+	testRegex := regexp.MustCompile(
 		`(?P<testName>dd[\w]*)` + // first capturing group: testfunc made of anything that isn't a whitespace
-		`(?:\s*)` + // match any whitespace (?: non capturing group)
-		`(?P<testFunc>[\S]*)` + // testfunc made of anything that isn't a whitespace
-		`(?:\s*\'?)` + // after can be any number of spaces and quotations if they exist (?: non capturing group)
-		`(?P<val1>\+?-?[^\t\f\v\' ]*)` + // first test val is anything that isnt a whitespace or a quoteation mark
-		`(?:'?\s*'?)` + // match any quotation marks and any space (?: non capturing group)
-		`(?P<val2>\+?-?[^\t\f\v\' ]*)` + // second test val is anything that isnt a whitespace or a quoteation mark
-		`(?:'?\s*'?)` +
-		`(?P<val3>\+?-?[^->]?[^\t\f\v\' ]*)` + //testvals3 same as 1 but specifically dont match with '->'
-		`(?:'?\s*->\s*'?)` + // matches the indicator to answer and surrounding whitespaces (?: non capturing group)
-		`(?P<expectedResult>\+?-?[^\r\n\t\f\v\' ]*)`) // matches the answer that's anything that is plus minus but not quotations
+			`(?:\s*)` + // match any whitespace (?: non capturing group)
+			`(?P<testFunc>[\S]*)` + // testfunc made of anything that isn't a whitespace
+			`(?:\s*\'?)` + // after can be any number of spaces and quotations if they exist (?: non capturing group)
+			`(?P<val1>\+?-?[^\t\f\v\' ]*)` + // first test val is anything that isnt a whitespace or a quoteation mark
+			`(?:'?\s*'?)` + // match any quotation marks and any space (?: non capturing group)
+			`(?P<val2>\+?-?[^\t\f\v\' ]*)` + // second test val is anything that isnt a whitespace or a quoteation mark
+			`(?:'?\s*'?)` +
+			`(?P<val3>\+?-?[^->]?[^\t\f\v\' ]*)` + //testvals3 same as 1 but specifically dont match with '->'
+			`(?:'?\s*->\s*'?)` + // matches the indicator to answer and surrounding whitespaces (?: non capturing group)
+			`(?P<expectedResult>\+?-?[^\r\n\t\f\v\' ]*)`) // matches the answer that's anything that is plus minus but not quotations
+
+	// Add regex to match to  rounding: rounding mode her
+
 	// capturing gorups are testName, testFunc, val1,  val2, and expectedResult)
-	ans := r.FindAllStringSubmatch(file, -1)
-	for _, a := range ans {
-		data = append(data, testCaseStrings{
-			testName:       a[1],
-			testFunc:       a[2],
-			val1:           a[3],
-			val2:           a[4],
-			val3:           a[5],
-			expectedResult: a[6],
-		})
+	ans := testRegex.FindStringSubmatch(line)
+
+	if len(ans) == 0 {
+		roundingRegex := regexp.MustCompile(`(?:rounding:[\s]*)(?P<rounding>[\S]*)`)
+		ans = roundingRegex.FindStringSubmatch(line)
+		if len(ans) == 0 {
+			return testCaseStrings{}, fmt.Errorf("No test cases")
+		}
+		return testCaseStrings{rounding: ans[1]}, fmt.Errorf("No test cases or rounding")
```

Comment:
This scenario seems to be "rounding but no test cases". The error message doesn't make this clear. Also, why do we need to make this distinction? The fact that there are no tests is the point of interest.

Reply:
No errors are returned anymore, but the calling function checks wether testCaseStrings.func is an empty string